src/native/external/zlib-ng/arch/arm/slide_hash_armv6.c

```diff
@@ -39,7 +40,8 @@ static inline void slide_hash_chain(Pos *table, uint32_t entries, uint16_t wsize
 }
 
 Z_INTERNAL void slide_hash_armv6(deflate_state *s) {
-    unsigned int wsize = s->w_size;
+    assert(s->w_size <= _UI16_MAX);
+    uint16_t wsize = (uint16_t)s->w_size;
```

Comment:
Unlike the deflate methods, in the arch/*/slide_hash*.c files, the `slide_hash_*` are always executed (do not depend on Debug) so I chose to call a real `assert` here.

I am hoping `<assert.h>` will be available in all these platforms. I could only verify arm64 and x86 locally. Probably `runtime-community` will pick up some if not all of the rest.